pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -430,11 +434,6 @@ def __validate_init_configuration(self) -> None:
                     f"ModelCheckpoint(save_top_k={self.save_top_k}, monitor=None) is not a valid"
                     " configuration. No quantity for top_k to track."
                 )
-            if self.save_last:
-                rank_zero_warn(
-                    "ModelCheckpoint(save_last=True, save_top_k=None, monitor=None) is a redundant configuration."
-                    " You can save the last checkpoint with ModelCheckpoint(save_top_k=None, monitor=None)."
-                )
```

Comment:
why's this being removed?

Reply:
As it stand, this warning is incorrect.
It is only correct if we make additional assumptions, for example every_n_epochs != 0.

For example:
`ModelCheckpoint(save_last=True, every_n_epochs=0, save_top_k=None, monitor=None)` only saves a ckpt at the end of training while `ModelCheckpoint(every_n_epochs=0, save_top_k=None, monitor=None)` does not save any ckpt.

Reply:
In that case, why not check it in the `if` condition?